binding.cc

```diff
@@ -1391,6 +1460,86 @@ NAPI_METHOD(db_get_many) {
   return promise;
 }
 
+/**
+ * Worker class for db.hasMany().
+ */
+struct HasManyWorker final : public PriorityWorker {
+  HasManyWorker(
+    napi_env env,
+    Database* database,
+    std::vector<std::string> keys,
+    napi_deferred deferred,
+    const bool fillCache
+  ) : PriorityWorker(env, database, deferred, "classic_level.has.many"),
+      keys_(std::move(keys)) {
+    options_.fill_cache = fillCache;
+    options_.snapshot = database->NewSnapshot();
+  }
+
+  void DoExecute () override {
+    cache_.reserve(keys_.size());
+
+    for (const std::string& key: keys_) {
+      std::string value;
+      leveldb::Status status = database_->Get(options_, key, value);
```

Comment:
It might be faster to use an iterator (if the LevelDB iterator lazily reads values thus allowing us to skip that; not sure) but it would add some complexity so I'll save that for a future PR (or for anyone who wants to take a stab at that).